### ISSUE TITLE
Why 'self.num_classes+1''???

### ISSUE THREAD

**Issue by Ordgod** [eriklindernoren/Keras-GAN#84]
in acgan.py:
```
 label = Dense(self.num_classes+1, activation="softmax")(features)
```
Why `self.num_classes+1` instead of `self.num_classe`???

I search this keyword `self.num_classes+1` in google and found that you are the first and only one to use it instead of `self.num_classe`. Could you please give a explanation?

**Comment by andrearama** [comment]
I agree with you, the discriminator output should give `self.num_classe` values and not `self.num_classe +1`. 
I explained what I think is the difference with the considered paper and why here: #85

**Comment by eriklindernoren** [comment]
I add an additional output for images produced by the generator. This will make sure that we don't force the discriminator to label fake images as coming from one of the class distributions.

**Comment by andrearama** [comment]
I understand your point of view, however I think that (1) this is not what the paper says and (2)  it is not the best way either, let me explain: 

(1) As I was also explaining in the pull request above-mentioned, the discriminator is supposed to give two 'independent' output describing the probability of the image of being fake and of belonging to a certain class (i.e. P(S | X) and P(C | X)). 
The paper explicitly says that `D is trained to maximize Ls + Lc while G is trained to maximize Lc − Ls`. In fact, both the networks have to maximize the Lc loss (given by P(C|X) ). This does not happen in the current implementation.

(2) In the current implementation, the generator receives a feedback from the discriminator less informative about the class of the generated image. That is because the part of the discriminator that should only give a probability distribution over the different classes is also including as "another class" the fake image case (which is superfluous as this feature is already treated in the other output and goes into Ls, but shouldn't be in Lc).

Hope I have not being too messy and you got my points

**Comment by andrearama** [comment]
Actually, the current implementation is more similar to another GAN architecture, that is the "Fast-converging Conditional Generative Adversarial Networks " : https://arxiv.org/abs/1805.01972

**Comment by eriklindernoren** [comment]
This has been fixed.
